### PR TITLE
[PT FE]: support aten::floor_divide_ and aten::movedim

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *aten::floor_divide*
 - *aten::movedim*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/frontends/pytorch/src/op/transpose.cpp [hunk 1]
```diff
@@ -81,7 +86,50 @@ OutputVector translate_t(const NodeContext& context) {
...
+    auto start = context.mark_node(v0::Constant::create(element::i32, {}, {0}));
+    auto step = context.mark_node(v0::Constant::create(element::i32, {}, {1}));
+    auto range = context.mark_node(std::make_shared<v4::Range>(start, rank, step, element::i32));
+    auto range_src = context.mark_node(std::make_shared<v4::Range>(start, rank, step, element::i32));
+    auto range_dst = context.mark_node(std::make_shared<v4::Range>(start, rank, step, element::i32));
+    auto dims_1d_shape = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {-1}));
+    // operation accepts 0d and 1d source and destination, make them always 1d
+    src_dims = context.mark_node(std::make_shared<v1::Reshape>(src_dims, dims_1d_shape, false));
+    dst_dims = context.mark_node(std::make_shared<v1::Reshape>(dst_dims, dims_1d_shape, false));
+    auto axis_0 = context.mark_node(v0::Constant::create(element::i32, Shape{}, {0}));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/transpose.cpp:112]: `start` and `axis_0` can be a single zero constant

File: src/frontends/pytorch/src/op/transpose.cpp [hunk 2]
```diff
@@ -81,7 +86,50 @@ OutputVector translate_t(const NodeContext& context) {
...
+        context.mark_node(std::make_shared<v3::ScatterElementsUpdate>(range_dst, dst_dims, minus_one_replaces, axis_0));
+    // Remove the dims whose position we already know, the ones marked with -1 in previous step
+    auto not_changed_src = context.mark_node(std::make_shared<v1::NotEqual>(src_perm_dims, dims_1d_shape));
+    auto not_changed_dst = context.mark_node(std::make_shared<v1::NotEqual>(dst_perm_dims, dims_1d_shape));
+    auto indices = context.mark_node(std::make_shared<v3::NonZero>(not_changed_dst, element::i32));
+    auto updates = context.mark_node(std::make_shared<v3::NonZero>(not_changed_src, element::i32));
+    // Update the position of the remaining dimensions. indices now contains the original position
+    // updates contains the new position it will shifted to after considering the user inputs.
+    indices = context.mark_node(std::make_shared<v1::Reshape>(indices, dims_1d_shape, false));
+    updates = context.mark_node(std::make_shared<v1::Reshape>(updates, dims_1d_shape, false));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/transpose.cpp:129]: Do we need these reshapes? It seems `indices` and `updates` will be created as 1d

File: src/frontends/pytorch/src/op/transpose.cpp [hunk 3]
```diff
@@ -81,7 +86,50 @@ OutputVector translate_t(const NodeContext& context) {
...
+    auto perm_dims = context.mark_node(std::make_shared<v3::ScatterElementsUpdate>(range, dst_dims, src_dims, axis_0));
+    auto src_perm_dims =
+        context.mark_node(std::make_shared<v3::ScatterElementsUpdate>(range_src, src_dims, minus_one_replaces, axis_0));
+    auto dst_perm_dims =
+        context.mark_node(std::make_shared<v3::ScatterElementsUpdate>(range_dst, dst_dims, minus_one_replaces, axis_0));
+    // Remove the dims whose position we already know, the ones marked with -1 in previous step
+    auto not_changed_src = context.mark_node(std::make_shared<v1::NotEqual>(src_perm_dims, dims_1d_shape));
+    auto not_changed_dst = context.mark_node(std::make_shared<v1::NotEqual>(dst_perm_dims, dims_1d_shape));
+    auto indices = context.mark_node(std::make_shared<v3::NonZero>(not_changed_dst, element::i32));
+    auto updates = context.mark_node(std::make_shared<v3::NonZero>(not_changed_src, element::i32));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/transpose.cpp:125]: Do we really need to calculate unchanged src dims and dst dims? Couldn't we just create tensor `[0, 1, 2, ..., rank-1]` and using `Select` take dims from this tensor to `perm_dims` on positions where `perm_dims` have `-1`?

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/frontends/pytorch/src/op/transpose.cpp (hunk 2 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/transpose.cpp:129]: nonzero returns 2d tensors

File: src/frontends/pytorch/src/op/transpose.cpp (hunk 3 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/transpose.cpp:125]: select provides different flow, I need shift dims that not affected, but select will keep them on the same places and will leads to repeating values

File: src/frontends/pytorch/src/op/transpose.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/transpose.cpp]: reused

File: src/frontends/pytorch/src/op/transpose.cpp [hunk 4]
```diff
@@ -81,7 +86,50 @@ OutputVector translate_t(const NodeContext& context) {
...
+    auto dst_dims = context.get_input(2);
+    Output<Node> rank;
+    std::tie(std::ignore, rank) = get_shape_rank(context, context.get_input(0), true);
+    src_dims = normalize_axis(context, src_dims, rank);
+    dst_dims = normalize_axis(context, dst_dims, rank);
+    auto start = context.mark_node(v0::Constant::create(element::i32, {}, {0}));
+    auto step = context.mark_node(v0::Constant::create(element::i32, {}, {1}));
+    auto range = context.mark_node(std::make_shared<v4::Range>(start, rank, step, element::i32));
+    auto range_src = context.mark_node(std::make_shared<v4::Range>(start, rank, step, element::i32));
+    auto range_dst = context.mark_node(std::make_shared<v4::Range>(start, rank, step, element::i32));
```
[the remark this answers is not in the document]
Comment (eaidova) [reply src/frontends/pytorch/src/op/transpose.cpp]: reused one

**Review by mvafin — APPROVED** [review]

**Pull request merged**